### DISCUSSION TITLE
Sendable & BindingAction protocol

### DISCUSSION THREAD

**Discussion by Alex0Klain** [pointfreeco/swift-composable-architecture#1938] [Q&A] [answered]
Hi, everybody!
I wanted to add a `Sendable` protocol to `Actions`, but I got a warning that `binding` is not signed under `Sendable`. Is this how it should be?
**Actions Code:**
```
public enum TestModuleActions: BindableAction, Equatable, Sendable {
  case onAppear
  case binding(BindingAction<TestModuleState>)
}
```
**State Code:**
```
public struct TestModuleState: Equatable, Sendable {
  public init() {}
}
```
**Warning:**
`Associated value 'binding' of 'Sendable'-conforming enum 'TestModuleActions' has non-sendable type 'BindingAction<TestModuleState>'`

**Comment by tgrapperon** [comment, marked as answer]
Hey @Alex0Klain! Because diagnostics were unreliable, the library is not quite yet committed regarding sendability requirements, and changes were postponed to a later time, but these changes are imminent if Xcode 14.3 enables sendability checks by default. 
In the meantime, you can import `ComposableArchitecture` like `@preconcurrency import ComposableArchitecture` and the warning will go away (I think that you can even narrow to `@preconcurrency import ComposableArchitecture.BindingAction` if you want to preserve concurrency warnings for other parts of the library). Your code is fine and the correct way to conform to `Sendable`.
When `BindingAction` will be sendable (it'll very likely be), you'll get a warning saying that `@preconcurrency` is not required anymore.

**Comment by stephencelis** [comment]
@Alex0Klain Also, if you are encountering some of these issues in the new Xcode beta, it appears that the sendability checking was accidentally made more strict by default, and will be made more lenient again soon: https://github.com/apple/swift/pull/63786
